Source/WebCore/rendering/MarkedText.h

```diff
@@ -68,24 +68,58 @@ struct MarkedText {
         Frontmost
     };
 
-    unsigned startOffset;
-    unsigned endOffset;
-    Type type;
+    MarkedText(unsigned startOffset, unsigned endOffset, Type type, const RenderedDocumentMarker* marker = nullptr, AtomString highlightName = { }, int priority = 0) : startOffset(startOffset), endOffset(endOffset), type(type), marker(marker), highlightName(highlightName), priority(priority) { };
+    MarkedText(WTF::HashTableDeletedValueType) : m_isHashTableDeletedValue(true) { };
+    MarkedText() { };
+
+    unsigned startOffset { 0 };
+    unsigned endOffset { 0 };
+    Type type { Type::Unmarked };
     const RenderedDocumentMarker* marker { nullptr };
     AtomString highlightName { };
-
+    int priority { 0 };
     bool isEmpty() const { return endOffset <= startOffset; }
+    bool isHashTableDeletedValue() const { return m_isHashTableDeletedValue; }
     bool operator==(const MarkedText& other) const
     {
-        return startOffset == other.startOffset && endOffset == other.endOffset && type == other.type && marker == other.marker && highlightName == other.highlightName;
+        return startOffset == other.startOffset && endOffset == other.endOffset && type == other.type && marker == other.marker && priority == other.priority
+        && ((highlightName == other.highlightName) || (highlightName.isNull() && other.highlightName.isNull()));
+    }
+    static unsigned hash(const MarkedText& key)
+    {
+        return key.startOffset ^ key.endOffset ^ static_cast<uint8_t>(key.type);
     }
 
     WEBCORE_EXPORT static Vector<MarkedText> subdivide(const Vector<MarkedText>&, OverlapStrategy = OverlapStrategy::None);
 
     static Vector<MarkedText> collectForDocumentMarkers(const RenderText&, const TextBoxSelectableRange&, PaintPhase);
     static Vector<MarkedText> collectForHighlights(const RenderText&, const TextBoxSelectableRange&, PaintPhase);
     static Vector<MarkedText> collectForDraggedContent(const RenderText&, const TextBoxSelectableRange&);
+
+private:
+    bool m_isHashTableDeletedValue { false };
```

Comment:
To avoid having an extra data member, we could just store `std::numeric_limits<unsigned>::max()` in startOffset as special value when it is the HashTableDeleted value. This is commonly done in WebKit.